libcxx/test/std/atomics/atomics.lockfree/is_always_lock_free.pass.cpp

```diff
@@ -45,7 +51,8 @@ void check_always_lock_free(std::atomic<T> const& a) {
 
   // In all cases, also sanity-check it based on the implication always-lock-free => lock-free.
   if (is_always_lock_free) {
-    std::same_as<bool> decltype(auto) is_lock_free = a.is_lock_free();
+    auto is_lock_free = a.is_lock_free();
+    ASSERT_SAME_TYPE(decltype(is_always_lock_free), bool const);
```

Comment:
```suggestion
    ASSERT_SAME_TYPE(decltype(is_lock_free), bool);
```

Reply:
Would you open a PR to fix this?

Reply:
https://github.com/llvm/llvm-project/pull/113169